source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -8,31 +8,49 @@ namespace RBAC {
 
 RoleBasedAccessControlEngineImpl::RoleBasedAccessControlEngineImpl(
     const envoy::config::filter::http::rbac::v2::RBAC& config, bool disabled)
-    : engine_disabled_(disabled),
-      allowed_if_matched_(disabled ||
-                          config.rules().action() ==
-                              envoy::config::rbac::v2alpha::RBAC_Action::RBAC_Action_ALLOW) {
-  if (disabled) {
-    return;
-  }
-
-  for (const auto& policy : config.rules().policies()) {
-    policies_.emplace_back(policy.second);
-  }
-}
+    : engine_disabled_(disabled), config_(config) {}
 
 RoleBasedAccessControlEngineImpl::RoleBasedAccessControlEngineImpl(
     const envoy::config::filter::http::rbac::v2::RBACPerRoute& per_route_config)
     : RoleBasedAccessControlEngineImpl(per_route_config.rbac(), per_route_config.disabled()) {}
 
 bool RoleBasedAccessControlEngineImpl::allowed(const Network::Connection& connection,
-                                               const Envoy::Http::HeaderMap& headers) const {
+                                               const Envoy::Http::HeaderMap& headers,
+                                               EnforcementMode mode) const {
   if (engine_disabled_) {
     return true;
   }
 
+  std::vector<PolicyMatcher> policies;
+  bool allowed_if_matched;
+  if (mode == EnforcementMode::ENFORCED) {
```

Comment:
the `if` and `else` is pretty similar, extract to a function.

Reply:
seems no extra function needed after moving code around :)